### PR TITLE
Home Page - Tests

### PR DESCRIPTION
**Pull request by ahmedmoeed**
Home page test cases. Check the structure and let me know so i can do the changes according to the practices you followed

### CODE REVIEW

File: pageobjects/homepage.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+var page = require('./page');
+var assert = require('assert');
+//var prop = require('./prop');
+
+//Page objects
+var saleButton = '//*[@class="b7afbb84"]'
+var locationBar = '//*[@placeholder="Location"]'
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: I think this locator will not work once you select any location filter. Can you verify this?

File: pageobjects/homepage.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+var page = require('./page');
+var assert = require('assert');
+//var prop = require('./prop');
+
+//Page objects
+var saleButton = '//*[@class="b7afbb84"]'
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: Using classes for location an element is not a good idea

File: pageobjects/homepage.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        //console.log(selectedbeds.getText() + ' is selected as the no. of beds')
+    }},
+
+    clickfindbutton : {get: function(){
+        $(findButton).click();
+        console.log(browser.getTitle());
+    }}
+})
+
+var homepageassertions = Object.create(page, {
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: I think there is no need to make another object. This method can be added on homepage object defined above

File: pageobjects/homepage.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+var page = require('./page');
+var assert = require('assert');
+//var prop = require('./prop');
+
+//Page objects
+var saleButton = '//*[@class="b7afbb84"]'
+var locationBar = '//*[@placeholder="Location"]'
+var locationIndex = '//li[@data-selected="true"]'
+var moreOption = '//span[@aria-label="Collapse expand button"]'
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: You can avoid adding a dependency of an element to be `span`. You could use `*` for it

File: pageobjects/homepage.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+var moreFilteroptions = '//span[@class="_514754a1"]'
+var propertyTypebox = '//div[@name="property type"]'
+var priceFilter = '//div[@aria-label="Price filter"]'
+var minPricerange = '//button[@aria-label="200,000"]'
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: Adding a selector for specific value is not a good idea.

File: pageobjects/homepage.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+var maxPricerange = '//div[@role="listbox"]//div[@class="ef391b9a"][2]'
+var bedsFilter = '//div[@aria-label="Bed filter"]'
+var selectedBeds = '//*[@aria-label="5"]'
+var findButton = '//a[@role="button"]'
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: This is working for now but its not a very same approach. We may add a new element on the home page with exactly same attributes.

File: specs/searchpage.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+
```
Comment (AdeelZahir) [inline specs/searchpage.js]: Select a name for spec file explaining what feature/use-case you are testing.

File: specs/searchpage.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+var homepage = require('../pageobjects/homepage');
+
+describe('homepage', function() {
+    it('should open the website and get title', function() {
```
Comment (AdeelZahir) [inline specs/searchpage.js]: It would be a good time to read mocha `it()` and `step()`

File: wdio.conf.js [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+    //     // maxInstances can get overwritten per capability. So if you have an in-house Selenium
+    //     // grid with only 5 firefox instances available you can make sure that not more than
+    //     // 5 instances get started at a time.
+    //     maxInstances: 5,
+    //     //
+    //     browserName: 'chrome'
+    // }],
+
+    capabilities: [
+        // {
```
Comment (AdeelZahir) [inline wdio.conf.js:51]: Add capabilities for chrome browser as well

File: wdio.conf.js [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+    // Test reporter for stdout.
+    // The only one supported by default is 'dot'
+    // see also: http://webdriver.io/guide/reporters/dot.html
+    // reporters: ['dot'],
+    //
+    // Options to be passed to Mocha.
+    // See the full list at http://mochajs.org/
+    mochaOpts: {
+        ui: 'bdd',
+        timeout: 99999999
```
Comment (AdeelZahir) [inline wdio.conf.js]: Update this timeout to be more realistic

**Review by AdeelZahir — CHANGES_REQUESTED** [review]
I have placed the comments as feedback. Please let me know if anything is unclear.

File: pageobjects/homepage.js (hunk 2 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Locator changed

File: pageobjects/homepage.js (hunk 3 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Additional method removed

File: pageobjects/homepage.js (hunk 4 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Dependency removed

File: pageobjects/homepage.js (hunk 5 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Specific value locator has been updated. Min and Max price functions are now dynamic

File: pageobjects/homepage.js (hunk 6 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Changed the logic and made it dynamic

File: wdio.conf.js (hunk 10 above)
Comment (ahmedmoeed) [reply wdio.conf.js]: Updated the timeout to 1 min

File: wdio.conf.js (hunk 9 above)
Comment (ahmedmoeed) [reply wdio.conf.js:51]: Capability added. But right now excluded the current spec.js file.

File: pageobjects/homepage.js (hunk 1 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Locator updated

File: pageobjects/homepage.js [hunk 11]
```diff
@@ -1,76 +1,106 @@
...
+            //var price = $(minPricerange)[i].getText()
+            if ($$(minPricerange)[i].getText() == minPrice)
+            {
+                $$(minPricerange)[i].click()
+                console.log(minPrice + ' is selected as the min price');
+            }
+        }
+    }},
+
+    selectMaxprice : {value: function(maxPrice){
```
Comment (AdeelZahir) [inline pageobjects/homepage.js]: `selectMaxprice` and `selectMinprice` methods are very similar. Can you make a single method for both cases?

File: wdio.conf.js [hunk 12]
```diff
@@ -94,7 +100,7 @@ exports.config = {
     baseUrl: 'https://bayut.com/',
     //
     // Default timeout for all waitFor* commands.
-    waitforTimeout: 99999999,
+    waitforTimeout: 1200000,
```
Comment (AdeelZahir) [inline wdio.conf.js]: `waitForTimeout` should not be greater than `mochaOpts.timeout`

Comment (ahmedmoeed) [reply wdio.conf.js]: waitforTimeout updated

File: pageobjects/homepage.js (hunk 11 above)
Comment (ahmedmoeed) [reply pageobjects/homepage.js]: Single Method used for both Min price and Max price

**Pull request merged**